components/menu/menu.tsx

```diff
@@ -0,0 +1,203 @@
+            event.preventDefault();
+
+            dispatch(openModal({
+                modalId: props.menuId,
+                dialogType: MenuModalComponent,
+            }));
+        }
+
+        return (
+            <button
+                id={props.triggerId}
+                aria-controls={props.menuId}
+                aria-haspopup='true'
+                aria-label={props.triggerAriaLabel}
+                className={props.triggerClassName}
+                onClick={handleAnchorButtonClickOnMobile}
+            >
+                {props.triggerElement}
+            </button>
+        );
+    }
+
+    function handleAnchorButtonClick(event: MouseEvent<HTMLButtonElement>) {
+        event.preventDefault();
+
+        setAnchorElement(event.currentTarget);
+    }
+
+    function rendertriggerElement() {
+        // Since the open and close state lies in this component, we need to force the visibility of the trigger element
+        const forceVisibleOnOpen = isMenuOpen ? {display: 'block'} : undefined;
+
+        const triggerElement = (
+            <button
+                id={props.triggerId}
+                aria-controls={isMenuOpen ? props.menuId : undefined}
+                aria-haspopup='true'
+                aria-expanded={isMenuOpen ? 'true' : undefined}
+                aria-label={props.triggerAriaLabel}
+                className={props.triggerClassName}
+                onClick={handleAnchorButtonClick}
+                style={forceVisibleOnOpen}
+                tabIndex={0}
+            >
+                {props.triggerElement}
+            </button>
+        );
+
+        if (props.triggerTooltipText) {
+            return (
+                <OverlayTrigger
+                    delayShow={OVERLAY_TIME_DELAY}
+                    placement={props?.triggerTooltipPlacement ?? 'top'}
+                    overlay={
+                        <Tooltip
+                            id={props.triggerTooltipId}
+                            className={props.triggerTooltipClassName}
+                        >
+                            {props.triggerTooltipText}
+                        </Tooltip>
+                    }
+                    disabled={!props.triggerTooltipText || isMenuOpen}
+                >
+                    {triggerElement}
+                </OverlayTrigger>
+            );
+        }
+
+        return triggerElement;
+    }
+
+    function handleMenuClose(event: MouseEvent<HTMLDivElement | HTMLUListElement>) {
+        event.preventDefault();
+        setAnchorElement(null);
+    }
+
+    return (
+        <CompassDesignProvider theme={theme}>
+            {rendertriggerElement()}
+            <MuiMenuStyled
+                id={props.menuId}
+                anchorEl={anchorElement}
+                open={isMenuOpen}
+                onClose={handleMenuClose}
+                aria-label={props.menuAriaLabel}
+                elevation={24}
+            >
+                <MuiMenuList
```

Comment:
I don't think we need to pass our own `MenuList` into the `Menu` when using MUI. That's probably why you're having to get rid of the extra padding below this

Reply:
Yes actually i read up a bit more, it seems MenuList would only be needed to if we are using custom Menu, so following that removed the component.